On off branch

zabu0003 opened the pull request:
Testing pull request by adding on off switch

zabu0003 committed:
Added switch functionality

zabu0003 committed:
Revision 2

LGKev commented:
@zabu0003  AWESOME JOB MAN. Really proud of you. This is a great skill to continue to learn and this is what people like to see. 

- need to set up an input in Setup()
- a normal while loop would work, but do while works too
- you mentioned possible comments and questions you had? can you put use // @kevin anywhere you had a question so i can search the file and put a better comment to answer your concern. 

by the way this way of sharing code is so much better than copying pasting. I can just review code instantly. 

one last thing I ask you is please test the code before we merge its how we prevent bugs from creeping in.

zabu0003 committed:
Revision and Comments added

Added pin input and two comments that need feedback before i modify anything else.

LGKev commented on EDL_Code.ino:
```diff
@@ -439,6 +441,14 @@ void straight(byte V_REF_L_VALUE, byte V_REF_R_VALUE){
+	This function stops both motors even if both encoders do not reach desired values.
+	Do we want seperate functions to stop each motor seperately so they both reach the desired encoder values,
+	or just modify this function with inputs of Left, Right, Both. Also we could slow one motors voltage to slow how
+	quickly it reaches its encoder value so that they reach them at similar times (a few options are available)
```
I agree that this could be improved. though this is only called after the encoder of the wheel is reached. if we were to use this to directly watch the vrefs and encoder counts you could do that. i think that it might be over kill. give it ago!

LGKev commented on EDL_Code.ino:
```diff
@@ -468,6 +478,14 @@ void loop(){
+	/*
+	ENCODER_L_COUNT_2_FEET_DISTANCE is smaller than ENCODER_R_COUNT_2_FEET_DISTANCE so while
+	loop stops both motors when left encoder reaches its value first, again a few options to fix depending
+	on how you want the program to function. Changing to right encoder and changing stop function, changing voltage of
```
I think thats a good idea! give it a shot. I am not sure how to do the slowing down bit. I was thinking we could collect data and and come up with an equation. mind you the arduino has limited math but i bet we could do an approximation of some kind. hopefully its linear. if not we could always do a look up table and a case statement. like

if encoder right is 40 ahead of the left and we want to stop soon then lower right Vref by 0.2 volts. 

or something. i'm not entirely sure how to do this.

LGKev replied:
also the while loop may just need to have 2 arguments. both left and right encoder. 

we should also see if the encoders are constantly the same distance apart. 
tomorrow i'm going to bring a bluetooth module so we can test faster.

no wires nothing in the way. the dc dc should have arrived and lipos are charged. we shoudl be able to test at full speed Vref = 255 

i like these concerns and how cool is github and git?
